src/dictionary.c

```diff
@@ -53,14 +51,20 @@ int Dictionary_Del(RedisModuleCtx *ctx, const char *dictName, RedisModuleString
     valuesDeleted += Trie_Delete(t, (char *)val, valLen);
   }
 
+  // Delete the dictionary if it's empty
+  if (t->size == 0) {
+    dictDelete(spellCheckDicts, dictName);
+    TrieType_Free(t);
+  }
+
   return valuesDeleted;
 }
 
-int Dictionary_Dump(RedisModuleCtx *ctx, const char *dictName, char **err) {
+void Dictionary_Dump(RedisModuleCtx *ctx, const char *dictName) {
   Trie *t = SpellCheck_OpenDict(ctx, dictName, REDISMODULE_READ);
   if (t == NULL) {
-    *err = "could not open dict key";
-    return -1;
+    RedisModule_ReplyWithSetOrArray(ctx, 0);
```

Comment:
This is not an error right? So need to fix the PR description - to just return an empty array.

Reply:
OK. PR description updated.